File: 0_1.baekjoon/brute-force/java/src/BOJ15651.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ15651 {
  
  private static int N, M;
  private static StringBuilder sb = new StringBuilder();

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    StringTokenizer st = new StringTokenizer(br.readLine());
    N = Integer.parseInt(st.nextToken());
    M = Integer.parseInt(st.nextToken());
    int[] arr = new int[M];

    dfs(arr, 0);
    System.out.println(sb.toString());
  }

  private static void dfs(int[] arr, int depth) {
    if (depth == M) {
      for (int num : arr) {
        sb.append(num).append(" ");
      }
      sb.append("\n");
      return;
    }

    for (int i = 1; i <= N; i++) {
      arr[depth] = i;
      dfs(arr, depth + 1);
    }

  }
}

File: 0_1.baekjoon/brute-force/kotlin/src/BOJ15651.kt
private val sb = StringBuilder()

fun main() = with(System.`in`.bufferedReader()) {
    val (n, m) = readLine().split(" ").map { it.toInt() }

    dfs(IntArray(m), 0, n, m)
    print(sb.toString())
}

private fun dfs(result: IntArray, depth: Int, n: Int, m: Int) {
    if (depth == m) {
        result.forEach { r -> sb.append(r).append(" ") }
        sb.append("\n")
        return
    }

    for (i in 1..n) {
        result[depth] = i
        dfs(result, depth + 1, n, m)
    }
}